Add cse for make_block_ptr in Triton codegen

nandesuka opened the pull request:
Summary: per title

Test Plan: added test cases

Differential Revision: D82648215




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

jansel reviewed (changes requested):
Failing tests?

njriasan reviewed (approved):
Looks good to me.

eellison reviewed (commented):
Looks good, but I think you just need to add the var shape, now PR requiring a shape for all CSEVariables has landed.

nandesuka committed:
Add cse for make_block_ptr in Triton codegen (#163399)

Summary:

per title

Test Plan: added test cases

Reviewed By: njriasan

Differential Revision: D82648215
